modules/cactus-web/src/helpers/state.ts

```diff
@@ -0,0 +1,24 @@
+import { SetStateAction, useCallback, useEffect, useRef, useState } from 'react'
+
+type StateSetterWithCallback<S> = (v: SetStateAction<S>, callback: () => void) => void
+
+const noop = () => undefined
+
+export const useStateWithSetterCallback = <S>(
+  initialState: S | (() => S)
+): [S, StateSetterWithCallback<S>] => {
+  const [state, setState] = useState<S>(initialState)
+  const storedCallback = useRef<() => void>(noop)
+
+  const setStateWithCallback = useCallback((v: SetStateAction<S>, callback: () => void) => {
```

Comment:
If you make the callback optional, this can be used in cases where sometimes you want a callback, and sometimes you don't.
```suggestion
  const setStateWithCallback = useCallback<StateSetterWithCallback<S>>((v, callback = noop) => {
```